### ISSUE TITLE
Canceling the input using сtrl+z resets the entire state of the editor

### ISSUE THREAD

**Issue by ekangash** [TypeCellOS/BlockNote#597]
**Describe the bug**
If, immediately after initializing the editor with an existing data set, we try to use the ctrl+z key combination, then the entire editor is reset to zero, regardless of which data set it was before.

This is inconvenient when there is an update of existing data and we are trying to roll back a new input, but in the end we get a complete reset of all data, which is confusing

**To Reproduce**
https://codesandbox.io/p/sandbox/overflow-editor-8mlw9y?file=%2Fsrc%2Findex.tsx

Is it possible not to reset the initial content? Well, either store the stack and roll back only the operations of the active session?

**Comment by asadbeksr** [comment]
Here is how you can handle it:
https://discord.com/channels/928190961455087667/1015169282444894219/1134448627096498217

You can make sure your handler runs before BlockNote's by making the event listener capture (add `true` as a third argument):
```ts
editor.domElement.addEventListener(
  "keydown", 
  cancelUndoCallback,
  true
);
```

**Comment by YousefED** [comment]
@ekangash I think this should be resolved in the latest version (0.12). Let me know if you still run into this after upgrading

**Comment by ekangash** [comment]
> @ekangash I think this should be resolved in the latest version (0.12). Let me know if you still run into this after upgrading

Indeed, the new update solved this problem, thank you very much.
